src/wp-includes/option.php

```diff
@@ -2144,20 +2144,47 @@ function update_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_update_site_option_{$option}", $value, $old_value, $option, $network_id );
 
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	global $wp_filter;
+
+	$pre_option_hook = is_multisite() ? "pre_site_option_{$option}" : "pre_option_{$option}";
+	if ( has_filter( $pre_option_hook ) ) {
```

Comment:
Following up to my feedback in https://github.com/WordPress/wordpress-develop/pull/5434#discussion_r1350417195, this still doesn't look right: For non-multisite, both the `pre_site_option_{$option}` _and_ `pre_option_{$option}` filters are applied here, so we need to take care of both.

However, at this point, I also wonder whether that is even the approach we should pursue, see @joemcgill's feedback in https://github.com/WordPress/wordpress-develop/pull/5434#discussion_r1350462532: Maybe we need functions to simply look up the raw value in the DB. Or more specifically to check whether an option exists in the DB, since that's the primary goal of this check.

However, similar to what happened in the other ticket 22192, I think there are arguments against any of those improvements that they could lead to BC breaks. Unfortunately, how we want this to work in an ideal case may not be possible without breaking BC for some specific scenarios, so we'll have to weigh these trade offs.